compiler/modules/front.syntax/parseStmt.bal

```diff
@@ -117,7 +117,15 @@ function parseStmt(Tokenizer tok) returns Stmt|err:Syntax {
             }
         }
         [DECIMAL_NUMBER, _]|[STRING_LITERAL, _]|"true"|"false" => {
-            return parseMethodCallStmt(tok);
+            if tok.peek() == "." || tok.peek() == "[" {
+                return parseMethodCallStmt(tok);
+            }
+            else {
+                return parseVarDeclStmt(tok, startPos);
+            }
+        }
+        [HEX_INT_LITERAL, _]| [DECIMAL_FP_NUMBER, _, _] => {
```

Comment:
I think this will have to be merged with above, we can have (semantically invalid) inputs like `0x2.toString()`. Please add that as test case.